src/spherical_mesh.cc

```diff
@@ -52,12 +38,17 @@ double SphericalMesh::volume(int i) const
 
 double SphericalMesh::gradient(int idx, double f_lo, double f_hi) const
     {
-    if (idx == 0 && origin_ == 0.){
+    if (idx == 0 && lower_ == 0.){
         return 0;
         }
     else{
         return (f_hi-f_lo)/(step_);
         }
     }   
-
+void SphericalMesh::validateBoundaryCondition()
+    {
```

Comment:
After making the `Mesh` implementation, you can call it first here then do the additional checks:
```suggestion
    {
    Mesh::validateBoundaryConditions();
```

Reply:
This still should be done here to make it explicit that this check is being called, see below.